astropy/coordinates/builtin_frames/icrs_cirs_transforms.py

```diff
@@ -20,22 +22,42 @@
 from .gcrs import GCRS
 from .cirs import CIRS
 from .hcrs import HCRS
+from .bcrs import BCRS
 from .utils import get_jd12, aticq, atciqz, get_cip, prepare_earth_position_vel
 
 
-# First the ICRS/CIRS related transforms
-@frame_transform_graph.transform(FunctionTransformWithFiniteDifference, ICRS, CIRS)
-def icrs_to_cirs(icrs_coo, cirs_frame):
-    # first set up the astrometry context for ICRS<->CIRS
+@frame_transform_graph.transform(FunctionTransform, ICRS, BCRS)
+def icrs_to_bcrs(icrs_coo, bcrs_frame):
+    if transform_propagator.get():
+        return transform_propagator.get().propagate(icrs_coo, bcrs_frame.obstime)
+    else:
+        data = icrs_coo.data
+        if data.isscalar and not bcrs_frame.obstime.isscalar:
```

Comment:
Yikes! This probably is the best argument yet for allowing attributes that have a bigger shape than the data, and then just broadcasting the data - but almost certainly best not in this PR! See #8819 for an earlier issue about this. I think we should fix this in the initializer...